datastore/query.go

```diff
@@ -241,6 +242,18 @@ func (q *Query) Offset(offset int) *Query {
 	return q
 }
 
+// BatchSize returns a derivative query to fetch the supplied number of results
+// at once. This value should be equal to or less than the Limit.
+func (q *Query) BatchSize(size int) *Query {
+	q = q.clone()
+	if size < math.MinInt32 || size > math.MaxInt32 {
```

Comment:
Should the check be `if size < 0 || size > math.MaxInt32`?

Reply:
Sorry for late reply.  Yes, it seems the negative values has no means.  I rewrite this.